services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py

```diff
@@ -191,23 +212,33 @@ async def start_study_job(
     webserver_api: Annotated[AuthSession, Depends(get_webserver_session)],
     director2_api: Annotated[DirectorV2Api, Depends(get_api_client(DirectorV2Api))],
     cluster_id: ClusterID | None = None,
-) -> JobStatus:
+) -> JobStatus | JSONResponse:
```

Comment:
Here I am a bit confused. do you need to do that with FastAPI? since it is pydantic based etc, why do you need to have a response with 2 different types?

Reply:
This is because when there is a conflict (meaning the computation was already started) then I return a 200 instead of a 202. So in other words, what changes between those two cases is not the model but the status code. I didn't see another way to do that than return the `JSONResponse` in the "special case" (conflict).

Reply:
I am also confused with these returns in handlers. IMO we should not annotate these returns provided that some magic conversions are generated by FastAPI based on `responses` and `response_model` arguments in the decorator. I think they confuse more than help

Reply:
OK, so you suggest I remove the type annotation. That I can do. But I don't get if I can avoid using the `JSONResponse`. The thing is that here we have 2 different successful responses, so I don't see how to handle that except for returning a JSONResponse.